### PR TITLE
early draft of pause functionality

### PR DESCRIPTION
**Pull request by matejdrazic**

### CODE REVIEW

File: src/bridgePausing.ts [hunk 1]
```diff
@@ -1,24 +1,50 @@
...
-      console.log("pero");
-      const response = await axios.get(ConfigUrl[environment]) as unknown as RawConfig;
-      console.log(response)
+      const network: keyof typeof ConfigUrl = configs.environment;
+      const {
+        privateKey,
+        mnemonic
+      } = configs;
+
+      const response = await axios.get(ConfigUrl[network]) as any;
```
Comment (nmlinaric) [inline src/bridgePausing.ts:33]: casting to `any` is a no go, leave ti like before `as unknown as RawConfig` and then you'll have proper type definiton

File: src/bridgePausing.ts [hunk 2]
```diff
@@ -1,24 +1,50 @@
...
+      const networks = response.data.domains.filter((network: any) => network.type === "evm"); // just evms for now
```
Comment (nmlinaric) [inline src/bridgePausing.ts:34]: Same for network, I think there is a type `Domain` in sygma SDK, import it from there

File: src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        const tx = await bridge.adminPauseTransfers();
+        const receipt = await tx.wait();
+        console.log(`Transaction no. ${i+1} completed, bridge on ${network.name} paused`);
+        receipts.push(receipt);
+    }
+    return receipts;
+}
+
+async function getBridgeContractInstance(network: any, wallet: ethers.Wallet) {
+    const bridge = new ethers.Contract(network.bridge, bridgeAbi, wallet);
```
Comment (nmlinaric) [inline src/utils.ts:50]: Actually we can avoid this importing of hardcoded ABI, if you import Bridge factory from Sygma SDK, check how it's done here:
https://github.com/sygmaprotocol/sygma-sdk/blob/82244855040a7835492263a19493b84e3becce86/packages/sdk/src/chains/EVM/assetTransfer.ts#L144

File: src/utils.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        const bridge = await getBridgeContractInstance(network, wallet)
+        const tx = await bridge.adminPauseTransfers();
+        const receipt = await tx.wait();
+        console.log(`Transaction no. ${i+1} completed, bridge on ${network.name} paused`);
+        receipts.push(receipt);
+    }
+    return receipts;
+}
+
+async function getBridgeContractInstance(network: any, wallet: ethers.Wallet) {
```
Comment (nmlinaric) [inline src/utils.ts:49]: no `any` :smiley:

File: src/utils.ts (hunk 3 above)
Comment (matejdrazic) [reply src/utils.ts]: Resolved

**Review by MakMuftic — CHANGES_REQUESTED** [review]
LGTM, @matejdrazic can we just add simple README instructions on running this command

**Pull request merged**